www/polj10/sp2/js/pastebin-rest.js

```diff
@@ -0,0 +1,531 @@
+    data.append("api_paste_expire_date", api_paste_expire_date);
+
+    postData(globals.create_paste_url, data,
+        function (responseText) {
+            const statusBox = document.getElementById("statusBox");
+            const statusLinkWrap = document.getElementById("statusLinkWrap");
+
+            if(responseText.includes("https://pastebin.com")){
+                const link = document.createElement("a");
+                link.title = responseText;
+                link.href = responseText;
+                link.innerText = responseText;
+                link.target = "_blank";
+                statusLinkWrap.innerHTML = "";
+                statusLinkWrap.appendChild(link);
+                statusBox.className = "";
+                statusBox.classList.add("green");
+
+                document.getElementById("pasteName").value = "";
+                document.getElementById("pasteTextArea").value = "";
+                listUserPastes();
+                return;
+            }
+
+            if(responseText === "Post limit, maximum pastes per 24h reached"){
+                showStatusPanel("crimson", "Dosáhl/a jste maximálního počtu vložených textů za 24 hodin.");
+                // statusLinkWrap.innerText = "Dosáhl/a jste maximálního počtu vložených textů za 24 hodin.";
+                // statusBox.className = "";
+                // statusBox.classList.add("crimson");
+                return;
+            }
+
+            if (responseText.includes("maximum")){
+                showStatusPanel("crimson", responseText);
+                // statusLinkWrap.innerText = responseText;
+                // statusBox.className = "";
+                // statusBox.classList.add("crimson");
+                return;
+            }
+
+            if(responseText.includes("empty")){
+                showStatusPanel("crimson", "Nebyl zadán žádný text.");
+                // statusLinkWrap.innerText = "Nebyl zadán žádný text.";
+                // statusBox.className = "";
+                // statusBox.classList.add("crimson");
+                return;
+            }
+
+            if(responseText.includes("expired")){
+                logout();
+                showStatusPanel("crimson", "Byl/a jste automaticky odhlášen/a");
+                // statusLinkWrap.innerText = "Byl/a jste automaticky odhlášen/a";
+                // statusBox.classList.add("crimson");
+                return;
+            }
+
+            showStatusPanel("crimson", "Při komunikaci se vyskytla chyba.");
+        },
+        function (statusText) {
+            showStatusPanel("crimson","Při komunikaci se vyskytla chyba.");
+        });
+}
+function createApiUserKey(userName, password) {
+    const data = new URLSearchParams();
+    data.append("api_dev_key", globals.api_dev_key);
+    data.append("api_user_name", userName);
+    data.append("api_user_password", password);
+
+    postData(globals.login_user_url, data,
```

Comment:
nedari se mi prihlasit se spravnymi udaji

Reply:
host musí být localhost, nebo fcp.vse.cz. Mám na své proxy také nastavené CORS :)
Psal jsem v PR message, že si nejsem jistý, jestli to mám pro fcp.vse.cz správně nastavené.

Reply:
SORRY! Zapomněl jsem, že jsem měl uvedenej v cors policy u localhostu nějakej divokej port. Dal jsem tam standartní "Live server" (z VS Code pluginu) port 5500.

Reply:
![image](https://user-images.githubusercontent.com/20724910/51125436-e6b2ac00-1820-11e9-8e81-be5ec71a11b9.png)

z localhostu bez portu mi to taky neslo :( nechces to udelat bez serveru? kdyztak existuji CORS proxy servery zadarmo

Reply:
tak uz to funguje

Reply:
OT: Nechtěl jsem používat CORS server třetí strany. Přeci jen by přes něj šlo heslo a user_api_key. 
Vyřešil jsem to tak, že jsem si napsal v ASP.NET Core backend, kterej kopíruje pastebin API a přeposílá vše, co pastebin odpoví.
Asi by to šlo udělat snadněji, ale pro mě bylo časově rychlejší to udělat takhle, než hledat alternativu.
A jako bonus nemám veřejnej pastebin "developer API key". Ten je uložený na backendu, se kterým komunikuju pomocí "vlastního api klíče".
Klient <--(Fake API key)--> pastebin.jiripolacek.net <--(Real API key)--> pastebin.com/api